activejob/lib/active_job/test_helper.rb

```diff
@@ -206,11 +250,12 @@ def assert_performed_with(args = {}, &_block)
         queue_adapter.performed_jobs = original_performed_jobs + performed_jobs
       end
 
-      def perform_enqueued_jobs
+      def perform_enqueued_jobs(only: nil)
         @old_perform_enqueued_jobs = queue_adapter.perform_enqueued_jobs
         @old_perform_enqueued_at_jobs = queue_adapter.perform_enqueued_at_jobs
         queue_adapter.perform_enqueued_jobs = true
         queue_adapter.perform_enqueued_at_jobs = true
+        queue_adapter.filter = only
```

Comment:
This leaks to subsequent test cases. It needs to be reset back to the original value in the `ensure` below.